suite-sparse: migrate from homebrew/science

fxcoudert opened the pull request:
See https://github.com/Homebrew/homebrew-core/issues/15619

ilovezfs commented on Formula/suite-sparse.rb:
```diff
@@ -0,0 +1,98 @@
+    # Only building libraries
+    system "make", "library", *make_args
+    system "make", "install", "INSTALL=#{prefix}", *make_args
+
```
This should probably be the end of the install method.

fxcoudert commented:
Updated to version 4.5.5. Removed `tbb` which is a mess to configure correctly, and is only used in one sublibrary. Do not build/install demos, but copy over a simple test.

ilovezfs commented on Formula/suite-sparse.rb:
```diff
@@ -0,0 +1,28 @@
+  url "http://faculty.cse.tamu.edu/davis/SuiteSparse/SuiteSparse-4.5.5.tar.gz"
+  sha256 "b9a98de0ddafe7659adffad8a58ca3911c1afa8b509355e7aa58b02feb35d9b6"
+
+  # SuiteSparse ships with metis5 but we use the Homebrew version
```
comment isn't needed

ilovezfs commented on Formula/suite-sparse.rb:
```diff
@@ -0,0 +1,28 @@
+  depends_on "metis"
+
+  def install
+    make_args = [ "BLAS=-framework Accelerate", "LAPACK=$(BLAS)",
```
this should be inlined since nothing is optionally appended to the array

fxcoudert replied:
We're using it twice

ilovezfs replied:
does make library install not work?

ilovezfs commented on Formula/suite-sparse.rb:
```diff
@@ -0,0 +1,28 @@
+                  "MY_METIS_LIB=-L#{Formula["metis"].opt_lib} -lmetis",
+                  "MY_METIS_INC=#{Formula["metis"].opt_include}" ]
+
+    # Only building libraries
```
comment not needed

ilovezfs commented on Formula/suite-sparse.rb:
```diff
@@ -0,0 +1,24 @@
+                   "MY_METIS_LIB=-L#{Formula["metis"].opt_lib} -lmetis",
+                   "MY_METIS_INC=#{Formula["metis"].opt_include}"
+
+    # Install a simple test
```
The comment probably isn't needed at all, but if you do want to keep it, it'll fit inline.
```
    pkgshare.install "KLU/Demo/klu_simple.c" # install a simple test 
```

ilovezfs commented on Formula/suite-sparse.rb:
```diff
@@ -0,0 +1,24 @@
+  test do
+    system ENV.cc, "-o", "test", pkgshare/"klu_simple.c", "-L#{lib}",
+                   "-I#{include}", "-lsuitesparseconfig", "-lklu"
+    system "./test"
```
anything we can verify here?

fxcoudert replied:
Not really, the output is numerical so subject to slight variations. I think compiling, linking and running is good enough.

ilovezfs replied:
ok

ilovezfs commented on Formula/suite-sparse.rb:
```diff
@@ -0,0 +1,23 @@
+                   "BLAS=-framework Accelerate", "LAPACK=$(BLAS)",
+                   "MY_METIS_LIB=-L#{Formula["metis"].opt_lib} -lmetis",
+                   "MY_METIS_INC=#{Formula["metis"].opt_include}"
+
```
this blank line can go away

fxcoudert replied:
You have too many style preferences :) you need to write a book

ilovezfs commented on Formula/suite-sparse.rb:
```diff
@@ -0,0 +1,23 @@
+  end
+
+  test do
+    system ENV.cc, "-o", "test", pkgshare/"klu_simple.c", "-L#{lib}",
```
would switch the order of "-I#{include}" and "-L#{lib}", though "-I#{include}" is not even needed

ilovezfs commented on Formula/suite-sparse.rb:
```diff
@@ -0,0 +1,26 @@
+      "INSTALL=#{prefix}",
+      "BLAS=-framework Accelerate", "LAPACK=$(BLAS)",
+      "MY_METIS_LIB=-L#{Formula["metis"].opt_lib} -lmetis",
+      "MY_METIS_INC=#{Formula["metis"].opt_include}"
```
I think it will require a comma at the end of this line

fxcoudert replied:
Nope, it works and audit is not complaining :)

ilovezfs replied:
Putting `"BLAS=-framework Accelerate", "LAPACK=$(BLAS)",` on a single line defeated the check, unfortunately

ilovezfs replied:
Fixed in https://github.com/Homebrew/homebrew-core/pull/15913.
